drivers/platform/stm32/stm32_dma.c

```diff
@@ -0,0 +1,284 @@
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include "stm32_dma.h"
+#include "no_os_error.h"
+#include "no_os_alloc.h"
+#include "stm32_irq.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Configure a DMA channel for a transfer.
+ * @param channel - The DMA channel descriptor.
+ * @param xfer - Descriptor for the transfer.
+ * @return 0 in case of success, negative error codes otherwise
+ */
+int stm32_dma_config_xfer(struct no_os_dma_ch *channel,
+			  struct no_os_dma_xfer_desc *xfer)
+{
+	struct stm32_dma_channel* sdma_ch;
+	int ret;
+
+	if (!channel || !xfer || !channel->extra)
+		return -EINVAL;
+
+	sdma_ch = channel->extra;
+
+	sdma_ch->hdma->Init.Channel = sdma_ch->ch_num;
+	sdma_ch->hdma->Init.MemInc = sdma_ch->mem_increment ? DMA_MINC_ENABLE :
+				     DMA_MINC_DISABLE;
+	sdma_ch->hdma->Init.PeriphInc = sdma_ch->per_increment ? DMA_PINC_ENABLE :
+					DMA_PINC_DISABLE;
+	sdma_ch->hdma->Init.MemDataAlignment = sdma_ch->mem_data_alignment;
+	sdma_ch->hdma->Init.PeriphDataAlignment = sdma_ch->per_data_alignment;
+	sdma_ch->hdma->Init.Mode = sdma_ch->dma_mode;
```

Comment:
don't really understand the purpose of these and also the switch statement below. basically they realize the same thing without the invalid value check

Reply:
Intention is to configure the DMA init parameters via the platform drivers. 
A switch case has been added since there are enum elements in the stm32_dma.h to realize the parameters, so that the user doesn't really have to look into the HAL drivers to configure the metrics. Do you suggest any other alternate approach here?

Reply:
what I mean is that you should remove the 3 lines above, because the switch blocks are assigning MemDataAlignment, PeriphDataAlignment and Mode fields.

Reply:
Oh yeah, you are right here. I had overlooked the duplicated assignments- I've removed the 3 lines above